vlm/src/main/scala/geotrellis/contrib/vlm/gdal/VSIPath.scala

```diff
@@ -0,0 +1,208 @@
+package geotrellis.contrib.vlm.gdal
+
+import scala.util.matching.Regex
+
+import java.io.File
+
+
+case class VSIPath(
+  path: String,
+  compressedFileDelimiter: String = "!"
+) {
+  import Schemes._
+  import Patterns._
+
+  private lazy val isLocal: Boolean =
+    schemeString.contains(FILE) || schemeString == ""
+
+  // Trying to read something locally on Windows matters
+  // because of how file paths on Windows are formatted.
+  // Therefore, we need to handle them differently.
+  private lazy val onLocalWindows: Boolean =
+    System.getProperty("os.name").toLowerCase == "win" && isLocal
+
+  lazy val scheme: Option[String] =
+    SCHEME_PATTERN.findFirstIn(path)
+
+  private lazy val schemeString: String =
```

Comment:
These kind of trivial field should always default to being a `def` while `lazy val` is "cheap" holding a trivial field on heap is kind of expensive. Rule of thumb I use is to treat `lazy val` as optimization, avoid it unless there is a specific reason to get the caching behavior.

Reply:
I see its used a lot though, so more of a case for `val`

Reply:
@jbouffard I'm with you, for re-use `val` makes sense. Please modify my comment towards as reaction towards seeing lots of `lazy` modifies.